### ISSUE TITLE
FindComputeCpp sets C++11 on host compiler overriding defaults

### ISSUE THREAD

**Issue by Ruyk** [codeplaysoftware/computecpp-sdk#18]
The FindComputeCpp module sets the C++11 flags on the host compiler, overriding the user settings:

 ```
# Set the host compiler C++ standard to C++11
set_property(TARGET ${targetName} PROPERTY CXX_STANDARD 11) 
```

This should only be set if there is not already a flag (or not set at all, and trust the user)

**Comment by mcopik** [comment]
I'd suggest not setting it all. It's the user responsibility and freedom to define flags other than including SYCL headers and linking to ComputeCpp library.

If the user makes the mistake of forgetting to set the standard, then compilation is going to fail in a rather obvious way (including headers requiring C++11, unknown keywords etc). If you are going to fix it for him and the user is supposed to built with C++14 or above, then you may create a more subtle bug. My suspicion is that most people are going to spend a lot of time analyzing their own building scripts until they start digging into third-party libraries.

**Comment by Ruyk** [comment]
+1

On the ComputeCpp side, we have a check for C++ support on the headers, so users can get a clear error message:

```
#if _MSC_FULL_VER < 170051025                                                   
#error "SYCL requires c++11 support"                                            
#endif                                                                          
                                                                                
#endif  // _MSC_FULL_VER                                                        
                                                                                
#else                                                                           
#if __cplusplus <= 199711L                                                      
#error "SYCL requires c++11 support"                                            
#endif                                                                          
#endif  // _MSC_VER  ```


So removing the flag from FindComputeCpp will not be an issue in terms of error messages.

**Comment by DuncanMcBain** [comment]
Hmmm. This breaks the samples though, and ideally I'd not replicate CXX_STANDARD across every single file. Maybe we could change the CMake config for the samples to do something like glob all the folders and look for the corresponding .cpp file? That way, we set it once.

The alternative that I had started working on was to query the language version, then if it was less than eleven, set it to eleven. Otherwise, just leave it alone.

**Comment by mcopik** [comment]
@DuncanMcBain I'm not sure but are you referring to setting CXX standard for each target?
https://cmake.org/cmake/help/v3.1/prop_tgt/CXX_STANDARD.html

You don't have to do it manually for each target because it is possible to change default target's property by setting this variable:
https://cmake.org/cmake/help/v3.1/variable/CMAKE_CXX_STANDARD.html

Changing value of this variable in samples/CMakeLists.txt should be sufficient.

**Comment by DuncanMcBain** [comment]
You're right! I'll make that fix. Thanks for your input, everyone!

**Comment by DuncanMcBain** [comment]
Fixed in f03ebbd. It looks like
```cmake
set_property(GLOBAL CXX_STANDARD 11)
```
doesn't work, even though the target version does. It works now, so whatever.
